kernel/nfft/nfft.c

```diff
@@ -2478,6 +2478,12 @@ static void nfft_adjoint_1d_B(X(plan) *ths)
 
 void X(trafo_1d)(X(plan) *ths)
 {
+  if((ths->N[0] <= ths->m) || (ths->n[0] <= 2*ths->m+2))
```

Comment:
Ok, the first condition makes sense, the second condition is not too clear to me. Theoretically, only `n > N` is required, so `n = N + 1` should be a valid choice. In fact, I tested this by modifying the 2D case in `simple_test.c` where I set `m = 7`, `N[1] = 8`, and `n[1] = 9`. I didn't spot any issues when running the transformation. This doesn't mean that there isn't any place in our code (maybe dependent on window function or pre-computation method) where we assume `2 * m + 2 < n`, but it's not obvious to me.